src/dataset.js

```diff
@@ -182,7 +203,8 @@ export default class Dataset {
       let state = this.state.update((next)=> {
         next.stats = stats;
         if(page !== next.pages[offset]) { return; }
-        next.pages[offset] = page.resolve(records);
+        // Filter on page update
+        next.pages[offset] = page.resolve(records).filter(this._filter);
```

Comment:
Should the filter be applied automatically when a page resolves? That way, when we want to reapply the filter to a resolved page, we just call `page.resolve` and the new filter is automatically reapplied!